### ISSUE TITLE
difficult to catch `Failed("DID NOT RAISE ...)` in `@pytest.mark.xfail(..., raises=...)`

### ISSUE THREAD

**Issue by graingert** [pytest-dev/pytest#8928]
#### What's the problem this feature will solve?
I'm currently trying to narrow an xfail: https://github.com/celery/celery/pull/6271/files#diff-ed1ef452735a964f7b23398a1833b2dcc22df133c2b06dca3ffba46fe842daf9R830

currently I need to use `raises=_pytest.main.Failed`

#### Describe the solution you'd like
I'd like `with pytest.raises(...):` to raise a public subclass of Failed or an AssertionError

I'm currently trying to narrow an xfail: https://github.com/celery/celery/pull/6271/files#diff-ed1ef452735a964f7b23398a1833b2dcc22df133c2b06dca3ffba46fe842daf9R830

**Comment by RonnyPfannschmidt** [comment]
its exposed as `pytest.fail.Exception` it ought to get a better one

**Comment by graingert** [comment]
TIL

**Comment by nicoddemus** [comment]
Close this?

**Comment by graingert** [comment]
>  it ought to get a better one

I think renaming/documenting pytest.fail.Exception makes sense here

**Comment by nicoddemus** [comment]
Agreed, but we should then either create a new issue or change the title of this one to reflect that. Personally I prefer creating a new one, with a clear description at the top, linking to this one.

**Comment by Zac-HD** [comment]
Closing as resolved: `_pytest.outcomes.Failed` is exposed as the public `pytest.fail.Exception`, and the agreed fix — documenting it — was done in #12285, which added `pytest.fail.Exception`/`skip.Exception`/`xfail.Exception` to the API reference, so `xfail(raises=pytest.fail.Exception)` is now a documented, supported pattern.
